kmir/src/kmir/kdist/mir-semantics/kmir.k

```diff
@@ -14,167 +15,717 @@ module KMIR-SYNTAX
   rule emptyOperands => .Operands
   syntax ProjectionElems ::= "noProjection" [function]
   rule noProjection => .ProjectionElems
-
-  syntax Value ::= FnVal( BasicBlocks, Int, Int )
+  syntax Map ::= "emptyMap" [function]
+  rule emptyMap => .Map
+  syntax VarDebugInfos ::= "noVarDebug" [function]
+  rule noVarDebug => .VarDebugInfos
 endmodule
 
-module KMIR
+module KMIR-MEMORY
+  imports INT
   imports KMIR-SYNTAX
   imports LIST
-  imports STRING
-  imports BOOL
-  imports BYTES
-  imports K-EQUAL
+  imports MAP
+
+  syntax Value
 
-  syntax CallStackItem  ::= CallRecord( FuncAddress, MaybeBasicBlockIdx, Place, UnwindAction, Values)
-  syntax FuncAddress    ::= FnAddr( Int )
-  syntax Values         ::= List{ Value, "" }
-  syntax Value          ::= // MIR Value Types
-                            Scalar( Int )
-                          | Ref( Int )
-                          | Ptr( Int, Value )
-                          | Array( Bytes )
-                          | Tuple( Values )
-                          | Closure( Int, Values )
-                          | Adt( Int, Int, Values )
-                            // Semantics Specific Types
-                          | Uninit()
+  configuration <memoryConfig>
+                  <body> .BasicBlocks </body>
+                  <stack> .List </stack>
```

Comment:
I would like to confirm with you, what is the index of the top of the stack? From what I read in the semantics, it seems to be 0. I refer to the K List data structure that we use as the stack, not the index of the base or local address that it holds. Followup: just to confirm: it seems that the indices in base, local addresses are the opposite? Is this so? Why?

Reply:
So, the current intended behavior is that we access the stack by counting from the bottom, not from the top.

Thus, if the stack height is N, then the bottom stack value has index 0 and the top value as index N-1.